src/GovUk.Education.ExploreEducationStatistics.Data.Api/Controllers/TableBuilderController.cs

```diff
@@ -13,22 +22,65 @@ namespace GovUk.Education.ExploreEducationStatistics.Data.Api.Controllers
     public class TableBuilderController : ControllerBase
     {
         private readonly ITableBuilderService _tableBuilderService;
+        private readonly IPersistenceHelper<ContentDbContext> _contentPersistenceHelper;
+        private readonly IUserService _userService;
 
-        public TableBuilderController(ITableBuilderService tableBuilderService)
+        public TableBuilderController(
+            ITableBuilderService tableBuilderService,
+            IPersistenceHelper<ContentDbContext> contentPersistenceHelper,
+            IUserService userService)
         {
             _tableBuilderService = tableBuilderService;
+            _contentPersistenceHelper = contentPersistenceHelper;
+            _userService = userService;
         }
-        
+
         [HttpPost]
         public Task<ActionResult<TableBuilderResultViewModel>> Query([FromBody] ObservationQueryContext query)
         {
             return _tableBuilderService.Query(query).HandleFailuresOrOk();
         }
 
         [HttpPost("release/{releaseId}")]
-        public Task<ActionResult<TableBuilderResultViewModel>> Query(Guid releaseId, [FromBody] ObservationQueryContext query)
+        public Task<ActionResult<TableBuilderResultViewModel>> Query(
+            Guid releaseId,
+            [FromBody] ObservationQueryContext query)
         {
             return _tableBuilderService.Query(releaseId, query).HandleFailuresOrOk();
         }
+
+        [ResponseCache(Duration = 300)]
+        [HttpGet("release/{releaseId}/datablock/{dataBlockId}")]
+        public async Task<ActionResult<TableBuilderResultViewModel>> QueryForDataBlock(
+            Guid releaseId,
+            Guid dataBlockId)
+        {
+            return await _contentPersistenceHelper.CheckEntityExists<ReleaseContentBlock>(
+                    query => query
+                        .Include(rcb => rcb.ContentBlock)
+                        .Include(rcb => rcb.Release)
+                        .Where(
+                            rcb => rcb.ReleaseId == releaseId
+                                   && rcb.ContentBlockId == dataBlockId
+                        )
+                )
+                .OnSuccessDo(block => this.CacheWithLastModified(block.Release.Published.GetValueOrDefault()))
```

Comment:
Do you need `GetValueOrDefault()` here since `CacheWithLastModified` has an optional last modified parameter. Perhaps this should be pushed down into `CacheWithLastModified` if the intention is never to set a response without a last modified header? Also I don't think it's a problem but the default value will be `01/01/0001 00:00:00` rather than null in case you weren't aware.

Reply:
Good point! I've removed the `GetValueOrDefault`. Think it was probably hangover from when I was initially putting the method together.